external/avro/src/main/scala/org/apache/spark/sql/avro/AvroSerializer.scala

```diff
@@ -234,14 +236,18 @@ class AvroSerializer(rootCatalystType: DataType, rootAvroType: Schema, nullable:
   }
 
   private def resolveNullableType(avroType: Schema, nullable: Boolean): Schema = {
-    if (nullable && avroType.getType != NULL) {
+    if (avroType.getType == Type.UNION && nullable) {
       // avro uses union to represent nullable type.
       val fields = avroType.getTypes.asScala
       assert(fields.length == 2)
       val actualType = fields.filter(_.getType != Type.NULL)
       assert(actualType.length == 1)
       actualType.head
     } else {
+      if (nullable) {
+        logWarning(s"Writing avro files with non-nullable avro schema with nullable catalyst " +
```

Comment:
nit. Shall we remove `s` at `s"` (line 248 and 249)?

Reply:
yes~ thanks.